### ISSUE TITLE
Training with orca.tf.estimator converges at a higher loss when compared to using TensorFlow/PyTorch with the same settings and data.

### ISSUE THREAD

**Issue by ottermegazord** [intel/analytics-zoo#559]
Using the same architecture, data and hyperparameters, I am able to replicate the performance of my TensorFlow model in PyTorch within a very small degree of variance. However the same implementation on Analytics Zoo yields a significantly higher loss function, even if the models are trained with the same inputs as the models built using TensorFlow and PyTorch. Is this expected?

Model Architecture
```
def model_zoo():
    """Autoencoder"""
    input_1 = tf.keras.Input(shape=(2016,), name='input_1')
    encoder_1 = tf.keras.layers.Dense(1024, name='encoder_1', activation='relu')(input_1)
    encoder_2 = tf.keras.layers.Dense(256, name='encoder_2', activation='relu')(encoder_1)
    encoder_3 = tf.keras.layers.Dense(128, name='encoder_3', activation='relu')(encoder_2)

    decoder_1 = tf.keras.layers.Dense(128, name='decoder_1', activation='relu')(encoder_3)
    decoder_2 = tf.keras.layers.Dense(256, name='decoder_2', activation='relu')(decoder_1)
    decoder_3 = tf.keras.layers.Dense(1024, name='decoder_3', activation='relu')(decoder_2)
    decoder_4 = tf.keras.layers.Dense(2016, name='decoder_4', activation='relu')(decoder_3)

    """DNN 3"""
    dense_3_1 = tf.keras.layers.Dense(128, name='dense_3_1', activation='relu')(encoder_3)
    dense_3_2 = tf.keras.layers.Dense(64, name='dense_3_2', activation='relu')(dense_3_1)
    dense_3_3 = tf.keras.layers.Dense(32, name='dense_3_3', activation='relu')(dense_3_2)
    dense_3_3 = tf.keras.layers.BatchNormalization()(dense_3_3)
    dense_3_3 = tf.keras.layers.Dropout(0.2)(dense_3_3)

    output = tf.keras.layers.Dense(1, name='output', activation='sigmoid')(dense_3_3)

    model = tf.keras.Model(inputs=[input_1],
                           outputs=[decoder_4, output])

    num_steps = 35

    learning_rate_fn = tf.keras.optimizers.schedules.PiecewiseConstantDecay(
        [50 * num_steps, 100 * num_steps, 30 * num_steps, 20 * num_steps, 50 * num_steps],
        [1e-2, 1e-3, 1e-4, 1e-5, 1e-6, 5e-6]
    )

    optimizer = tf.keras.optimizers.Adam(learning_rate=learning_rate_fn)

    model.compile(
        optimizer=optimizer,
        loss={
            'decoder_4': 'mse',
            'output': 'binary_crossentropy',
        }
        # metrics=['accuracy']
    )

    return model
```
